backend/src/libs/packages/encrypt/encrypt.service.ts

```diff
@@ -0,0 +1,31 @@
+import bcrypt from 'bcryptjs';
+
+class EncryptService {
+  private saltRounds: number;
+
+  public constructor(saltRounds: number) {
+    this.saltRounds = saltRounds;
+  }
+
+  public async generateSalt(saltRounds?: number): Promise<string> {
+    if (saltRounds) {
+      return await bcrypt.genSalt(saltRounds);
+    }
+    if (this.saltRounds) {
+      return await bcrypt.genSalt(this.saltRounds);
+    }
+    return await bcrypt.genSalt();
```

Comment:
```suggestion
    return await bcrypt.genSalt(saltRounds ?? this.saltRounds);
```
We will always have this.saltRounds, because the argument is required in constructor

Reply:
It should be required or I can change the constructor to make it optional?

Reply:
I prefer to have it required